Sources/Parseable.swift

```diff
@@ -39,18 +39,46 @@ public extension Parseable {
         }
     }
 
-    /// Get a json Dictionary back
+    /// Get a json Dictionary back one level deep. Meaning we do not Parse relations.
+    /// if you want your relations parsed implement `CustomSerializable`
     public var json: [String: Any?] {
-        var internalMap = [String: Any]()
-        let mirror = Mirror(reflecting: self)
-        for child in mirror.children {
-            internalMap[child.label!] = child.value
+
+        if let customSelf = self as? CustomSerializable {
+            var internalMap = [String: Any]()
+            let mirror = Mirror(reflecting: self)
+            for child in mirror.children where customSelf.isRelation(for: child.label!) == false{
+                internalMap[child.label!] = child.value
+            }
+
+            for child in mirror.children where customSelf.isRelation(for: child.label!) == true {
+                let relation = customSelf.jsonForRelation(with: child.label!)
+                switch relation {
+                case .nodeArray(let array):
+                    internalMap[child.label!] = array
+                case .nodeObject(let object):
+                    internalMap[child.label!] = object
+                default:
+                    break
+                }
+            }
+            return internalMap
+        } else {
+            var internalMap = [String: Any]()
+            let mirror = Mirror(reflecting: self)
+            for child in mirror.children {
+                internalMap[child.label!] = child.value
+            }
+            return internalMap
         }
-        return internalMap
     }
     
 }
 
+public protocol CustomSerializable {
```

Comment:
❓Maybe comment the code with some examples, because I find the function naming a bit to complex.